lib/charms/blackbox_k8s/v0/blackbox_probes.py

```diff
@@ -0,0 +1,652 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Blackbox Probes Library.
+
+## Overview
+
+This document explains how to integrate with the Blackbox charm
+for the purpose of providing a probes metrics endpoint to Prometheus.
+
+## Provider Library Usage
+
+The Blackbox Exporter charm interacts with its datasources using this charm
+library. The goal of this library is to be as simple to use as possible.
+Charms seeking to expose metric endpoints to be probed via Blackbox, must do so
```

Comment:
Afaiu, there are no metric endpoints to begin with, which is why we're using blackbox exporter (?).